minetest/minetest
# Replace Irrlicht file picker with new Lua dialog

Based on #10580, this PR replaces the current Irrlicht file/folder picker with a new one written in Lua. The code for the old Irrlicht picker dialog on the Minetest engine side of things has also been removed.

Fixes #8073

![image](https://github.com/minetest/minetest/assets/60856959/2f40581e-1bbd-4a08-ac04-be1ff284b6a5)

## To do
This PR is a Ready for Review.

## How to test
- Test with a setting expecting a filepath (e.g. `font_path`). Check so that you can only pick files, not folders.
- Test with a setting expecting a (folder)path (e.g. `shader_path`). Check so that you can only pick folders, not folders.

Platforms:
- Linux: Works for me
- Android: Barely functions due to it not being possible to double tap a formspec table (but does work with a keyboard, haha). Not a regression from the old Irrlicht dialog though, which straight out wouldn't function on Android.
- Windows: Needs testing.
